book/src/gobject_values.md

```diff
@@ -17,16 +17,32 @@ enum Value <T> {
     // boxed types
     String(Option<String>),
     Object(Option<T: impl IsA<Object>>),
-}
 ```
 
 That means that a `glib::Value` can represent a set of types, but only one of them per instance.
-Also note that boxed types such as `String` or `Object` are wrapped in an `Option`.
+For example, this is how you would use a `Value` representing an `i32`.
+
+<span class="filename">Filename: listings/gobject_values/1/main.rs</span>
+
+```rust,no_run
+{{#rustdoc_include ../listings/gobject_values/1/main.rs:i32}}
+```
+
+Please note that boxed types such as `String` or `Object` are wrapped in an `Option`.
 This comes from C, where boxed types can always be `None` (or null in C terms).
-That also means that the retrieval of Rust types from `glib::Value` differs slightly between boxed and non-boxed types.
+If you are sure that your boxed type, will never be `None`, you can just access it the same way as with the `i32` above.
+
+<span class="filename">Filename: listings/gobject_values/1/main.rs</span>
+
+```rust,no_run
+{{#rustdoc_include ../listings/gobject_values/1/main.rs:string}}
+```
+
+In case your `Value` might be `None`, just call `get::<Option<T>>` instead.
+If you do that, it will give you the chance to check the output instead of panicking immediately.
```

Comment:
It's not panicking. It's returning an `Err` just like with type mismatches :)

Reply:
In my code example it would panic, because I call expect. 
I agree that the current wording is confusing though.

Reply:
I gave it another try